### PR TITLE
Multicast http handlers

### PR DESCRIPTION
**Pull request by michaelfig**
Separate endpoints into `/private/*` (only accessible from localhost) and public ones.

The support in this PR for `sendMulticast` (no global broadcast anymore from the HTTP vat) anticipates #748

### CODE REVIEW

**Comment by michaelfig** [comment]
The PR also moves and refactors:

* `/vat` (REPL support) into `/private/repl`
* `/captp` (CapTP support) into `/private/captp`

File: packages/cosmic-swingset/lib/ag-solo/web.js [hunk 1]
```diff
@@ -101,24 +108,21 @@ export function makeHTTPListener(basedir, port, host, rawInboundCommand) {
...
-  // todo: later allow arbitrary endpoints?
-  for (const ep of ['/vat', '/wallet-public', '/api']) {
-    app.post(ep, (req, res) => {
-      if (ep !== '/api' && !validateOrigin(req)) {
-        res.json({ ok: false, rej: 'Invalid Origin' });
-        return;
-      }
+  // accept POST messages to arbitrary endpoints
+  app.post('*', (req, res) => {
+    if (!validateOrigin(req)) {
```
Comment (warner) [inline packages/cosmic-swingset/lib/ag-solo/web.js:113]: as before, I want to consider origin checks to be weak, and build a proper auth mechanism soon

**Review by warner — APPROVED** [review]
I wasn't able to look at this too closely, but in a quick skim, nothing jumped out as wrong.

File: packages/agoric-cli/agoric-wallet-build/asset-manifest.json [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
-    "main.js.map": "./static/js/main.49aad819.chunk.js.map",
+    "main.js": "./static/js/main.873f8d69.chunk.js",
+    "main.js.map": "./static/js/main.873f8d69.chunk.js.map",
     "runtime-main.js": "./static/js/runtime-main.6d19b3b8.js",
     "runtime-main.js.map": "./static/js/runtime-main.6d19b3b8.js.map",
     "static/js/2.80be0d87.chunk.js": "./static/js/2.80be0d87.chunk.js",
     "static/js/2.80be0d87.chunk.js.map": "./static/js/2.80be0d87.chunk.js.map",
     "index.html": "./index.html",
-    "precache-manifest.8f442f599464d859b54c66608d0d6853.js": "./precache-manifest.8f442f599464d859b54c66608d0d6853.js",
+    "precache-manifest.c6e69f21c2aea8705c2c098601680840.js": "./precache-manifest.c6e69f21c2aea8705c2c098601680840.js",
```
[the remark this answers is not in the document]
Comment (michaelfig) [reply packages/agoric-cli/agoric-wallet-build/asset-manifest.json:10]: I'm thinking about it.  The only other way I know is to break out wallet-frontend into its own repo, then publish it (with generated files) on NPM, which agoric-cli can rely on.  Is that acceptable?

File: packages/agoric-cli/lib/deploy.js [hunk 3]
```diff
@@ -30,7 +30,7 @@ export default async function deployMain(progname, rawArgs, powers) {
     ws.send(JSON.stringify(obj));
   };
 
-  const wsurl = `ws://${hostport}/captp`;
+  const wsurl = `ws://${hostport}/private/captp`;
```
[the remark this answers is not in the document]
Comment (michaelfig) [reply packages/agoric-cli/lib/deploy.js:33]: Yes, that's where we're going.  Webkey-style things would be best.  The current mechanism is totally bogus, but no more totally bogus than other same-origin-policy designs.

File: packages/cosmic-swingset/package.json [hunk 4]
```diff
@@ -30,6 +30,7 @@
     "@agoric/registrar": "^0.0.1",
     "@agoric/same-structure": "^0.0.1",
     "@agoric/sharing-service": "^0.0.1",
+    "@agoric/spawner": "^0.0.3",
```
[the remark this answers is not in the document]
Comment (michaelfig) [reply packages/cosmic-swingset/package.json:33]: It was, but cosmic-swingset is not yet lint-clean, so the CI skips it.

File: packages/cosmic-swingset/lib/ag-solo/web.js (hunk 1 above)
Comment (michaelfig) [reply packages/cosmic-swingset/lib/ag-solo/web.js:113]: Absolutely!

File: packages/agoric-cli/agoric-wallet-build/asset-manifest.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (warner) [reply packages/agoric-cli/agoric-wallet-build/asset-manifest.json:10]: Could we generate those files during a build step in agoric-cli? (at the expense of requiring folks run `yarn build` after making changes to anything under `agoric-wallet-build/`)

**Comment by michaelfig** [comment]
Sounds good, I'll implement that the next time I need to change the wallet.

-- 
Michael FIG
Software Engineer, Agoric

**Pull request merged**